### DISCUSSION TITLE
.Net 9 Frame becoming Obsolete

### DISCUSSION THREAD

**Discussion by Shannon-roos** [PrismLibrary/Prism#3292] [.NET MAUI]
With Frame being marked as obsolote in .Net 9 is there an alternative for a Dialog View? 

I did try using ContentPage but Prism doesn't like regisering that as a Dialog... 

The type 'MAUI.Dialogs.Views.DatePickerDialogView' cannot be used as type parameter 'TView' in the generic type or method 'DialogRegistrationExtensions.RegisterDialog<TView, TViewModel>(IContainerRegistry, string)'. There is no implicit reference conversion from 'MAUI.Dialogs.Views.DatePickerDialogView' to 'Microsoft.Maui.Controls.View'.

Just wondering if I am doing something wrong or if there will be a change?

**Comment by dansiegel** [comment]
I'm not sure what your issue is exactly a Dialog can be any type of .NET MAUI View including a StackLayout, Grid, or a Border which happens to be the replacement for a Frame. The `IDialogContainer` is a Page. By default this would be a ContentPage provided by Prism which is pushed as a modal. The Popups Plugin implements this with a PopupPage.

**Comment by Shannon-roos** [comment]
My Apologies, thats actually what I understood as well and tried it with Border, however VS errored about not being able to use the "Attached Propery Resources" on a Border so I swapped it back and parked it.

Just tried it again with Border as the root and restarted VS, (which honestly I should know better to do by now) and its fine. 

Thanks for clarifying...
